### PR TITLE
[Merged by Bors] - chore: move definition of ENat and PNat out of Order.TypeTags

### PR DESCRIPTION
**Pull request by kim-em**
These had been moved to `Order.TypeTags` to reduce imports elsewhere, but they are sort of off-topic here, so I've just moved them to their own files.

### CODE REVIEW

**Comment by kim-em** [comment]
(I decided to do this after having to delete these sections while doing a manual minimization.)

File: Mathlib/Data/ENat/Basic.lean [hunk 1]
```diff
@@ -3,12 +3,13 @@ Copyright (c) 2022 Yury Kudryashov. All rights reserved.
...
-import Mathlib.Algebra.CharZero.Lemmas
-import Mathlib.Algebra.Order.Ring.WithTop
-import Mathlib.Algebra.Order.Sub.WithTop
+import Mathlib.Data.ENat.Defs
 import Mathlib.Data.Nat.Cast.Order.Basic
 import Mathlib.Data.Nat.SuccPred
 import Mathlib.Order.Nat
+import Mathlib.Algebra.CharZero.Lemmas
+import Mathlib.Algebra.Order.Ring.WithTop
+import Mathlib.Algebra.Order.Sub.WithTop
```
Comment (Vierkantor) [inline Mathlib/Data/ENat/Basic.lean:12]:
```suggestion
import Mathlib.Algebra.CharZero.Lemmas
import Mathlib.Algebra.Order.Ring.WithTop
import Mathlib.Algebra.Order.Sub.WithTop
import Mathlib.Data.ENat.Defs
import Mathlib.Data.Nat.Cast.Order.Basic
import Mathlib.Data.Nat.SuccPred
import Mathlib.Order.Nat
```

File: Mathlib/Data/ENat/Defs.lean [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Simon Hudon, Yury Kudryashov
+-/
+
+import Mathlib.Order.TypeTags
+import Mathlib.Data.Nat.Notation
```
Comment (Vierkantor) [inline Mathlib/Data/ENat/Defs.lean:8]:
```suggestion
import Mathlib.Data.Nat.Notation
import Mathlib.Order.TypeTags
```

**Review by Vierkantor — APPROVED** [review]
LGTM, thanks!

bors d+

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**